airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -868,10 +784,131 @@ private List<MetricAttribute> generateMetricAttributes(final ConnectionUpdaterIn
+      // Act as a return
+      prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+    }
+
+    return true;
+  }
+
+  private GeneratedJobInput generateJobInput(final ConnectionUpdaterInput connectionUpdaterInput) {
+    workflowInternalState.setJobId(getOrCreateJobId(connectionUpdaterInput));
+    workflowInternalState.setAttemptNumber(createAttempt(workflowInternalState.getJobId()));
+
+    return getJobInput();
+  }
+
+  private void checkConnectionAndRunSync(final ConnectionUpdaterInput connectionUpdaterInput, final GeneratedJobInput jobInputs) {
+    reportJobStarting();
+
+    try {
+      final SyncCheckConnectionFailure syncCheckConnectionFailure = checkConnections(jobInputs);
+      if (syncCheckConnectionFailure.isFailed()) {
+        workflowState.setSyncOutput(syncCheckConnectionFailure.buildFailureOutput());
+        workflowState.setFailed(getFailStatus(workflowState.getStandardSyncOutput()));
+        workflowState.setFailureCause(FailureCause.CONNECTION);
+      } else {
+        workflowState.setSyncOutput(runChildWorkflow(jobInputs));
+        workflowState.setFailed(getFailStatus(workflowState.getStandardSyncOutput()));
+
+        if (workflowState.isFailed()) {
+          workflowState.setFailureCause(FailureCause.UNKNOWN);
+        }
+      }
+
+    } catch (final ChildWorkflowFailure childWorkflowFailure) {
+      // when we cancel a method, we call the cancel method of the cancellation scope. This will throw an
+      // exception since we expect it, we just
+      // silently ignore it.
+      if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
+        // do nothing, cancellation handled by cancellationScope
+        recordWorkflowFailureCountMetric(connectionUpdaterInput, FailureCause.CANCELED);
+      } else if (childWorkflowFailure.getCause()instanceof final ActivityFailure af) {
+        // Allows us to classify unhandled failures from the sync workflow. e.g. If the normalization
+        // activity throws an exception, for
+        // example, this lets us set the failureOrigin to normalization.
+        workflowInternalState.getFailures().add(FailureHelper.failureReasonFromWorkflowAndActivity(
+            childWorkflowFailure.getWorkflowType(),
+            af.getActivityType(),
+            af.getCause(),
+            workflowInternalState.getJobId(),
+            workflowInternalState.getAttemptNumber()));
+        workflowState.setFailureCause(FailureCause.ACTIVITY);
+        workflowState.setFailed(true);
+      } else {
+        workflowInternalState.getFailures().add(
+            FailureHelper.unknownOriginFailure(childWorkflowFailure.getCause(), workflowInternalState.getJobId(),
+                workflowInternalState.getAttemptNumber()));
+        workflowState.setFailureCause(FailureCause.WORKFLOW);
+        workflowState.setFailed(true);
+      }
+    }
+  }
+
+  private void reportJobStatus(final ConnectionUpdaterInput connectionUpdaterInput) {
```

Comment:
I find the flow unclear.

What do you think of changing `void reportJobStatus` to `bool handleTermination`? My understanding of this part is that we check if we should stop the job more than just reporting status.
The piece that is also not super clear to me is why we call `prepareForNextRUnAndContinueAsNew` from here.
I think the flow would be clearer if the main function would look like
```
if (handleTermination(connectionUpdaterInput)) {
    prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
    return;
}
```

Reply:
I like this idea too.

Currently written, it's confusing because the method is called `reportJobStatus`, and majority of the method does report job status, however at the end of this the prepare for next run method is called and that is super important to this running.

Reply:
done (Need to push)